src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -295,7 +295,8 @@ private MethodInfo CreateFunctionTarget(CancellationToken cancellationToken)
                 ICompilation compilation = _compilationService.GetFunctionCompilation(Metadata);
                 FunctionSignature functionSignature = compilation.GetEntryPointSignature(_functionEntryPointResolver);
 
-                ValidateFunctionBindingArguments(functionSignature, throwIfFailed: true);
+                ImmutableArray<Diagnostic> bindingDiagnostics = ValidateFunctionBindingArguments(functionSignature, throwIfFailed: true);
```

Comment:
That's correct. Validation still triggered a compilation exception in the case of errors. We were just missing the traces in diagnostics with lower levels (e.g. Warning)